src/App.tsx

```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isSetReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return (!isSetReversed) ? visibleGoods : visibleGoods.reverse();
+}
+
 export const App: React.FC = () => {
+  const {
+    ALPHABET,
+    LENGTH,
+    NONE,
+  } = SortType;
+
+  const [sortType, setSortType] = useState(NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+  const setResetting = (type: SortType, reverse: boolean) => {
```

Comment:
```suggestion
  const reset = () => {
```
Can't you just reset the sortType and reverse without any parameters?